### PR TITLE
feature/SC-5674-refactoringHomework

### PR DESCRIPTION
**Pull request by TomaszJuchimiuk**
# Description

  I transferred all methods from homework.js to new PO. I matched the test. It is not finished yet. My plan is to merge my changes to develop, than I have to wait for Alicja refactoring of course area (courses and homeworks are connected, I don't want to work in this area in the same time to avoid conflicts), then I will back to finish refactoring the homework/tasks area.

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by yuliashikhareva** [comment]
Let's set WIP label on this PR.

File: page-objects/pages/HMWRKHomeworkListPage.js [hunk 1]
```diff
@@ -1 +1,119 @@
...
+		let sortBtn = await driver.$(
+			'#filter > div > div.md-chip.md-theme-default.md-deletable.md-clickable > div'
+		);
+		await sortBtn.click();
+		let select = await driver.$('#selection-picker > div > div');
+		await select.click();
+		let lastedited = await driver.$('body > div.md-select-menu.md-menu-content-bottom-start.md-menu-content-small.md-menu-content.md-theme-default > div > ul > li:nth-child(2) > button'
+		);
+		await lastedited.click();
+		let ok = await driver.$('.md-button.md-primary.md-theme-default > div > div');
```
Comment (kgorner) [inline page-objects/pages/HMWRKHomeworkListPage.js:26]: please use await waitHelpers.waitAndClick(ok);

File: page-objects/pages/HMWRKHomeworkListPage.js [hunk 2]
```diff
@@ -1 +1,119 @@
...
+	await driver.pause(1000);
+	},
+
+	sortHometasks: async function() {
+		let sortBtn = await driver.$(
+			'#filter > div > div.md-chip.md-theme-default.md-deletable.md-clickable > div'
+		);
+		await sortBtn.click();
+		let select = await driver.$('#selection-picker > div > div');
+		await select.click();
```
Comment (kgorner) [inline page-objects/pages/HMWRKHomeworkListPage.js:22]: please use please use await waitHelpers.waitAndClick(select);

File: page-objects/pages/HMWRKHomeworkListPage.js [hunk 3]
```diff
@@ -1 +1,119 @@
...
+	},
+	areThereAnyTasks: async function() {
+		let elementWithTasks = await driver.$$('.col-xl-12');
+		return elementWithTasks.length > 0 ? true : false;
+	},
+	chooseTaskAmongAllTasks: async function(taskname) {
+		let taskindex = await this.returnTaskIndex(taskname);
+		if(taskindex!=false) {
+			let task = await driver.$('.col-xl-12 > li:nth-child('+taskindex+') > .content > h2');
+			await task.click();
```
Comment (kgorner) [inline page-objects/pages/HMWRKHomeworkListPage.js:54]: please use await waitHelpers.waitAndClick(task);

File: page-objects/pages/HMWRKHomeworkPage.js [hunk 4]
```diff
@@ -1 +1,74 @@
...
+module.exports = {
+// student helpers
+userFindsTheTask: async function(taskname) {
+    let areThereAnyTasks = await driver.$$('#homeworks > ol > div > li');
+    await expect(areThereAnyTasks.length).not.to.equal(0);
+    for (var i=1; i<=areThereAnyTasks.length; i++) {
+        let taskSelector = await driver.$('#homeworks > ol > div > li:nth-child('+i+') .h5.title');
+        let tasknameOnPage = await taskSelector.getText();
+        if(tasknameOnPage==taskname){
+            await taskSelector.click();
```
Comment (kgorner) [inline page-objects/pages/HMWRKHomeworkPage.js:18]: please use await waitHelpers.waitAndClick(taskSelector);

File: page-objects/pages/HMWRKHomeworkPage.js [hunk 5]
```diff
@@ -1 +1,74 @@
...
+            await driver.pause(1000);
+        }
+    }
+},	
+switchToSubmissionTab: async function() {
+    let submissionTab = "#submission-tab-link";
+    await helpewaitHelpersrs.waitAndClick(submissionTab);
```
Comment (kgorner) [inline page-objects/pages/HMWRKHomeworkPage.js:25]: helpewaitHelpersrs ==> waitHelpers

File: page-objects/pages/HMWRKHomeworkPage.js [hunk 6]
```diff
@@ -1 +1,74 @@
...
+},
+submitSolutionForTheHometask: async function() {
+    await driver.switchToFrame(0);
+    let iframeBody = await driver.$('body');
+    let assignmentText = 'here is some text which I want to submit';
+    await iframeBody.setValue(assignmentText);
+    await driver.switchToParentFrame();
+    let container = await driver.$('#submission');
+    let submitBtn = await container.$('button[type="submit"]');
+    await submitBtn.click();
```
Comment (kgorner) [inline page-objects/pages/HMWRKHomeworkPage.js:35]: please use await waitHelpers.waitAndClick(submitBtn);

File: page-objects/pages/HMWRKHomeworkPage.js [hunk 7]
```diff
@@ -1 +1,74 @@
...
+    await evaluationTab.click();
+    let evaluation = await driver.$(courseData.elem.evaluationInProcent);
+    await evaluation.setValue(95);
+    await driver.switchToFrame(0);
+    let body = await driver.$('body');
+    let comment = 'sehr gut!';
+    await body.setValue(comment);
+    await driver.switchToParentFrame();
+},
+}
```
Comment (kgorner) [inline page-objects/pages/HMWRKHomeworkPage.js:74]: please add a new line at the end of the file (see the little red minus)
please format the document (right-click inside your open document in VS Code) > format document

File: src.code-workspace [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+	"folders": [
+		{
+			"path": "..\\..\\Qiagen\\dPCR-software-suit"
+		}
+	]
+}
```
Comment (kgorner) [inline src.code-workspace:7]: please add a new line at the end of the file (see the little red minus)
please format the document (right-click inside your open document in VS Code) > format document

File: src.code-workspace [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (kgorner) [inline src.code-workspace:1]: Are you sure we need this in our workspace-file?

**Comment by TomaszJuchimiuk** [comment]
new PR https://github.com/hpi-schul-cloud/integration-tests/pull/122 to this branch.

**Pull request closed without merging**